### PR TITLE
[5.x] Implies only the new migrations behaviour on L11

### PR DESCRIPTION
**Pull request by nunomaduro**
This pull request makes the new behaviour (regarding applying the database migrations on the selected) a thing only in Laravel 11.

### CODE REVIEW

File: src/NewCommand.php [hunk 1]
```diff
@@ -465,7 +465,7 @@ protected function promptForDatabaseOptions(string $directory, InputInterface $i
                 default: $defaultDatabase
             );
 
-            if ($database !== $defaultDatabase) {
+            if ($database !== $defaultDatabase && $this->hasMariaDBConfig($directory)) {
```
Comment (driesvints) [inline src/NewCommand.php:468]: @nunomaduro I wonder if we should add a new `isLaravel11OrHigher` method or something similar to better indicate the reason for this check?

File: src/NewCommand.php [hunk 2]
```diff
@@ -308,22 +308,18 @@ protected function configureDefaultDatabaseConnection(string $directory, string
...
-        }
+        $version = json_decode(file_get_contents($directory.'/composer.json'), true)['require']['laravel/framework'];
+        $version = str_replace('^', '', $version);
+        $version = explode('.', $version)[0];
 
-        return str_contains(
-            file_get_contents($directory.'/config/database.php'),
-            "'mariadb' =>"
-        );
+        return $version >= 11;
```
[the remark this answers is not in the document]
Comment (nunomaduro) [reply src/NewCommand.php:322]: That function don't work when the version is something like 10.0 without patch version. So, it's not worth the complexity IMO.

[the remark this answers is not in the document]
Comment (driesvints) [reply src/NewCommand.php:322]: Think it would work if it was just `'11'` but it's fine indeed.

https://3v4l.org/a5F7O

**Comment by taylorotwell** [comment]
I don't think we'll take any action here on L10 for now. If anything, we would backport our fix from L11.

**Review by driesvints — APPROVED** [review]

**Pull request closed without merging**